extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -333,18 +348,54 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
       dev().warn(TAG, `Frame already exists, sra: ${this.useSra}`);
       return '';
     }
+
+    const ampRtcPageElement = document.getElementById('amp-rtc');
+    let rtcRequestPromise = null;
+    let rtcConfig;
+    if (ampRtcPageElement) {
+      rtcConfig = tryParseJson(ampRtcPageElement.innerHTML);
+      if (rtcConfig && typeof rtcConfig == 'object'
+      && !!rtcConfig['doubleclick']) {
+        rtcRequestPromise = this.sendRtcRequestPromise_(/** @type {!Object} */(
+            rtcConfig['doubleclick']));
+      }
+    }
     // TODO(keithwrightbos): SRA blocks currently unnecessarily generate full
     // ad url.  This could be optimized however non-SRA ad url is required to
     // fallback to non-SRA if single block.
     this.populateAdUrlState();
     // TODO: Check for required and allowed parameters. Probably use
     // validateData, from 3p/3p/js, after noving it someplace common.
     const startTime = Date.now();
-    return getPageLevelParameters_(this.win, this.getAmpDoc(), startTime)
-        .then(pageLevelParameters =>
-        googleAdUrl(this, DOUBLECLICK_BASE_URL, startTime,
-            Object.assign(this.getBlockParameters_(), pageLevelParameters),
-          ['108809080']));
+    const pageLevelParametersPromise = getPageLevelParameters_(
+        this.win, this.getAmpDoc(), startTime);
+    const blockParameters = this.getBlockParameters_();
```

Comment:
You don't actually need blockParameters for RTC.  You only need populateAdUrlState to have set this.jsonTargeting.  I suggest we structure as follows:
- getPageParams
- populateAdUrlState
- sendRtcRequest which modifies this.jsonTargeting_ and return an object mapping to be merged into the end url.  sendRtcRequest contains ALL RTC logic (perhaps rename to executeRTC?) from merge to timeout to reject if pub indicates no ad request on failure
- getBlockParams (AFTER RTC completion)
- Object.assign block, RTC, page (note that order indicates order of url parameters!)